diff --git a/third_party/blink/renderer/core/dom/element.cc b/third_party/blink/renderer/core/dom/element.cc
index 42be0451fd7b1..11615f5627a38 100644
--- a/third_party/blink/renderer/core/dom/element.cc
+++ b/third_party/blink/renderer/core/dom/element.cc
@@ -7770,6 +7770,11 @@ ShadowRoot* Element::OpenShadowRoot() const {
   return root && root->GetMode() == ShadowRootMode::kOpen ? root : nullptr;
 }
 
+ShadowRoot* Element::FakeShadowRoot() const {
+  ShadowRoot* root = GetShadowRoot();
+  return root;
+}
+
 ShadowRoot* Element::ClosedShadowRoot() const {
   ShadowRoot* root = GetShadowRoot();
   if (!root) {
diff --git a/third_party/blink/renderer/core/dom/element.h b/third_party/blink/renderer/core/dom/element.h
index 451d392c1c608..98d729c8b370c 100644
--- a/third_party/blink/renderer/core/dom/element.h
+++ b/third_party/blink/renderer/core/dom/element.h
@@ -1098,6 +1098,7 @@ class CORE_EXPORT Element : public ContainerNode {
     return HasShadowRoot() ? GetShadowRootInternal() : nullptr;
   }
   ShadowRoot* OpenShadowRoot() const;
+  ShadowRoot* FakeShadowRoot() const;
   ShadowRoot* ClosedShadowRoot() const;
   ShadowRoot* AuthorShadowRoot() const;
   ShadowRoot* UserAgentShadowRoot() const;
diff --git a/third_party/blink/renderer/core/dom/element.idl b/third_party/blink/renderer/core/dom/element.idl
index 51f29e18f378f..cd36caa0d7d87 100644
--- a/third_party/blink/renderer/core/dom/element.idl
+++ b/third_party/blink/renderer/core/dom/element.idl
@@ -81,6 +81,7 @@ dictionary ParseHTMLUnsafeOptions {
 
     [RaisesException, MeasureAs=ElementAttachShadow] ShadowRoot attachShadow(ShadowRootInit shadowRootInitDict);
     [PerWorldBindings, ImplementedAs=OpenShadowRoot] readonly attribute ShadowRoot? shadowRoot;
+    [PerWorldBindings, ImplementedAs=FakeShadowRoot] readonly attribute ShadowRoot? fakeShadowRoot;
 
     [RaisesException] Element? closest(DOMString selectors);
     [RaisesException] boolean matches(DOMString selectors);
